### ISSUE TITLE
Rotate/flip extension which keeps the fill angle

### ISSUE THREAD

**Issue by tyrosinase** [inkstitch/inkstitch#3080]
When you mirror or rotate an object in Inkscape, the fill angle stays the same relative to the frame. I'm guessing there's no hook that lets Ink/Stitch know the action happened. But could we have a menu option (a la re-stack) that does so within Ink/Stitch?

Being able to mirror and rotate on save would be even better (although possibly for only one user: me) if it was also available from the command line. And maybe scale, while I'm wishlisting - right now I'm doing that in two steps, one Inkscape command-line re-save of a temp SVG that's scaled, and one Ink/Stitch command to get the complete ZIP. Scale is less of a priority since it's doable from Inkscape, but mirror only works for things with default fills, and rotate only for things with contour.

(For some reason I've been doing a lot of designs with mirrored pieces lately: the moths, and right now I have some things that would be tiny enough for earrings and I'm realizing those should be mirrored.)

![20240611_174459](https://github.com/user-attachments/assets/924f5461-3289-43ad-a321-62fc5caab1b0)

**Comment by kaalleen** [comment]
That might be a perfect Situation for the usage oft Clones? You can even clone the hole group.

**Comment by capellancitizen** [comment]
Yeah, the revamped clones should let you clone the whole thing as a group in one go and mirror it, and the fill angle(s) should be properly mirrored on the cloned element(s).

(Now that I think about it, if you need to rotate or mirror an object but want to keep the relative fill angles, you can clone it, transform the clone however you want, then unlink the clone and delete the original. Might be a cool trick to document somewhere, it's probably handy for some niche cases but it isn't something that's immediately obvious...)

**Comment by tyrosinase** [comment]
Oh, hmm, thanks. I wonder if I can manage that on the command line. 🤔

**Comment by capellancitizen** [comment]
You can, but it's a bit clunky. An example:
```
SELF_CALL=true inkscape -g --batch-process --actions="file-open:drawing-4.svg;window-open;select-by-id:layer1;clone;select-by-selector:svg>use;org.d-inkstitch.unlink-clone.noprefs;select-clear;select-by-id:layer1;delete;export-filename:out.svg;export-do"
```

**Comment by Marif91** [comment]
Je pense que je règlerais cette situation avec le plan de broderie. Si je veux un deuxième élément entièrement symétrique, j'établis le plan de broderie, c'est cet objet que je rendrai symétrique et je suis sûre que l'angle des points sera bon....

**Comment by kaalleen** [comment]
Maybe we can add mirror functionality to the panelization option in the zip export.

**Comment by kaalleen** [comment]
@tyrosinase  I wonder if it really makes sense for panelization or not. Maybe the given methods are sufficient already?

**Comment by tyrosinase** [comment]
> [@tyrosinase](https://github.com/tyrosinase) I wonder if it really makes sense for panelization or not. Maybe the given methods are sufficient already?

It would be a nice-to-have but since it's possible (thanks to @capellancitizen for saving me the scavenger-hunt of finding out how to do it on the command line, btw) it's probably not worth it unless it's trivial to add.

Although: it might be nicer-to-have an Ink/Stitch-specific version of rotate/flip (like we have for Break Apart) that maintains the stitch angles (and anything else that might come up). Might be more general-purpose than putting it in panelization, though certainly nontrivilal. I have just learned to wait to set my stitch angles until I'm done arranging my objects, so it's still not that critical, but it might be more newbie-friendly?

**Comment by kaalleen** [comment]
Interesting idea!
